base/operators.jl

```diff
@@ -248,7 +261,7 @@ function setindex_shape_check(X::AbstractArray, I...)
     i = j = 1
     while true
         ii = size(X,i)
-        jj = length(I[j])::Int
+        jj = I[j]
```

Comment:
Since this function used to take the indexes themselves rather than their sizes, I wonder if we should change the signature to `I::Int...`. There might have been some callers outside of base.